lib/ansible/modules/files/file.py

```diff
@@ -169,7 +172,7 @@
   file:
     path: /etc/some_directory
     state: directory
-    mode: 0755
+    mode: '0755'
```

Comment:
Well, the 0 is not longer essential for it to work (while it was if there were no quotes) but the leading zero is essential to understand it is an octal number. So personally, I would avoid any confusion by leaving it in place for the examples, force strings and don't mention anything else.

If we force strings (#50035) we avoid any issues and if the zero is left out (by accident) it would still work.

Reply:
Okay, let's leave the leading zeros on the examples. And perhaps update the shared docs for `mode` again, depending on what happens with #50035.